src/libQuiet/Decoder.js

```diff
@@ -0,0 +1,124 @@
+import { FRAME_BUFFER_SIZE, SAMPLE_BUFFER_SIZE } from '../constants';
+
+import createTranslator from './createTranslator';
+import libQuietProvider from './libQuietProvider';
+
+export default class Decoder {
+    constructor(profile, sampleRate, enableStats) {
+        this._decoder = createTranslator('decoder', profile, sampleRate);
+
+        if (enableStats) {
+            libQuietProvider.get()
+                .ccall(
+                    'quiet_decoder_enable_stats',
+                    null,
+                    [ 'pointer' ],
+                    [ this._decoder ]
+                );
+        }
+    }
+
+    consume(samples) {
+        libQuietProvider.get()
+            .ccall(
+                'quiet_decoder_consume',
+                'number',
+                [ 'pointer', 'pointer', 'number' ],
+                [ this._decoder, samples.getRawSamples(), SAMPLE_BUFFER_SIZE ]
+            );
+    }
+
+    destroy() {
+        libQuietProvider.get()
+            .ccall(
+                'quiet_decoder_destroy',
+                null,
+                [ 'pointer' ],
+                [ this._decoder ]
+            );
+
+        this._decoder = null;
+    }
+
+    getChecksumFails() {
+        return libQuietProvider.get()
+            .ccall(
+                'quiet_decoder_checksum_fails',
+                'number', 
+                [ 'pointer' ],
+                [ this._decoder ]
+            );
+    }
+
+    getReceiverStats() {
+        const libQuiet = libQuietProvider.get();
+        const numFramesPtr = libQuiet.ccall(
+            'malloc',
+            'pointer',
+            [ 'number' ],
+            [ 4 ]
+        );
+        const frames = libQuiet.ccall(
+            'quiet_decoder_consume_stats',
+            'pointer',
+            [ 'pointer', 'pointer' ],
+            [ this._decoder, numFramesPtr ]
+        );
+
+        // time for some more pointer arithmetic
+        const numFrame = libQuiet.HEAPU32[numFramesPtr / 4];
+
+        libQuiet.ccall(
+            'free',
+            null,
+            [ 'pointer' ],
+            [ numFramesPtr ]
+        );
+
+        // Why these numbers?
+        const framesize = 4 + 4 + 4 + 4 + 4;
```

Comment:
I believe this is sizeof() of this struct
https://github.com/quiet/quiet/blob/master/include/quiet.h#L543

The _really_ nice way to fix this would be for us to add another .c file that gets compiled into the emscripten bundle that can answer questions like "What's the size of a `quiet_decoder_frame_stats`"?